src/fcntl.rs

```diff
@@ -574,6 +578,18 @@ pub fn fcntl(fd: RawFd, arg: FcntlArg) -> Result<c_int> {
                 *path = PathBuf::from(OsString::from(optr.to_str().unwrap()));
                 return Ok(ok_res)
             },
+            #[cfg(target_os = "freebsd")]
+            F_KINFO(path) => {
+                let mut info: libc::kinfo_file = std::mem::zeroed();
+                info.kf_structsize = std::mem::size_of::<libc::kinfo_file>() as i32;
+                let res = libc::fcntl(fd, libc::F_KINFO, &mut info);
+                let ok_res = Errno::result(res)?;
+                let sl = info.kf_path.to_vec();
```

Comment:
> This heap allocation (`to_vec()`) seems to be unnecessary and can be replaced with the following approach:
> 
> ```rust
> let path = info.kf_path;
> let u8_slice = slice::from_raw_parts(path.as_ptr().cast(), path.len());
> let optr = CStr::from_bytes_until_nul(u8_slice).unwrap();
> ```

I did so because I thought there was a lifetime issue but whatever approach I take, the test always works locally.

Reply:
From CI, it seems that the code can compile without any issue, so I think there is no lifetime problem.

BTW, any idea why the test is failing on the FreeBSD CI?

```
failures:

---- test_fcntl::test_f_kinfo stdout ----
thread 'test_fcntl::test_f_kinfo' panicked at 'assertion failed: `(left == right)`
  left: `""`,
 right: `"/tmp/.tmpkEJvT4"`', test/test_fcntl.rs:599:5


failures:
    test_fcntl::test_f_kinfo

test result: FAILED. 201 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 1.67s

error: test failed, to rerun pass `--test test`

Exit status: 101
```

Reply:
What I was trying to say :-) I battled the whole evening yesterday, no problem locally but always CI failure.

Reply:
Oh, I get it now.

If you are debugging this stuff, it might be faster to create a GitHub repo just containing this patch and set a FreeBSD 14 CI from Cirrus since it will have ONLY one CI to run (nix has a lot)